### PR TITLE
Change related to selecting app service plan type for dev and non-dev environments

### PR DESCRIPTION
**Pull request by manojvazirani**
Change related to selecting app service plan type for dev and non-dev environments:
If is_dev is true, it will select the S1 plan else will select S3 plan
Details for S1 and S3 can be seen in the link below:
https://azure.microsoft.com/en-us/pricing/details/app-service/linux/

No tests have been performed except ARM validation, since release pipelines are only running on master. If the dev deployment goes through without any issues, we can promote it to go to staging and prod.

### CODE REVIEW

File: infrastructure/azuredeploy.json [hunk 1]
```diff
@@ -6,6 +6,14 @@
       "type": "string",
       "defaultValue": "[uniqueString(resourceGroup().id)]"
     },
+    "dev_env": {
+      "type": "string",
```
Comment (bengreenier) [inline infrastructure/azuredeploy.json]: type: "boolean" might be more applicable here

Comment (manojvazirani) [reply infrastructure/azuredeploy.json]: Done.

**Comment by manojvazirani** [comment]
@allydurks @bengreenier : Sorry, you guys will have to review again as I made changes after running validation on the ARM template.

**Comment by manojvazirani** [comment]
Tested by deploying a dev and non-dev environment which deployed successfully

**Pull request merged**